int32 player_id
int32 action_type
int32 action_value
